baseplate/observers/tracing.py

```diff
@@ -539,18 +540,24 @@ class TraceQueueFullError(Exception):
 
 
 class SidecarRecorder(Recorder):
-    """Interface for recording spans to a POSIX message queue.
+    """Interface for recording spans to a message queue.
 
     The SidecarRecorder serializes spans to a string representation before
     adding them to the queue.
     """
 
-    def __init__(self, queue_name: str):
-        self.queue = MessageQueue(
-            "/traces-" + queue_name,
-            max_messages=MAX_QUEUE_SIZE,
-            max_message_size=MAX_SPAN_SIZE,
-        )
+    def __init__(self, queue_name: str, use_in_memory_queue: bool = False):
+        if use_in_memory_queue:
```

Comment:
Like Fred said, using a queue type argument would be better. You could define and [enum](https://docs.python.org/3/library/enum.html?highlight=enum#module-enum) in the `message_queue` module that could be used everywhere to refer to the various options.